src/test/java/seedu/address/logic/parser/AddModuleParserTest.java

```diff
@@ -1,174 +1,160 @@
-        //         new AddCommand(expectedPerson));
-
+    public void parse_multipleTags_success() {
+        Module expectedModule = new ModuleBuilder().withName(VALID_MODULENAME_CS2103T)
+                .withMC(VALID_MC_4).withTag(VALID_TAG_TUTORIAL).withTag(VALID_TAG_LECTURE).build();
+        assertParseSuccess(parser, multipleTags, new AddModuleCommand(expectedModule));
     }
 
     @Test
     public void parse_compulsoryNameFieldMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddModuleCommand.MESSAGE_USAGE);
-        String missingName = " " + PREFIX_ZOOM_LINK + VALID_ZOOMLINK_CS2103T
-                + " " + PREFIX_MODULAR_CREDITS + VALID_MC_4;
-        // missing name prefix
+        String missingName = " " + PREFIX_MODULAR_CREDITS + VALID_MC_4 + " " + PREFIX_TAG + VALID_TAG_LECTURE;
         assertParseFailure(parser, missingName, expectedMessage);
     }
 
     @Test
     public void parse_compulsoryAllFieldsMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddModuleCommand.MESSAGE_USAGE);
         String missingAll = "";
-        // all prefixes missing
         assertParseFailure(parser, missingAll, expectedMessage);
     }
 
     @Test
     public void parse_invalidName_failure() {
-        //String invalidName = " " + PREFIX_NAME + "@123" + " " + PREFIX_ZOOM_LINK + "www.example.com";
-        // invalid name
-        //assertParseFailure(parser, invalidName, ModuleName.MESSAGE_CONSTRAINTS);
-    }
-
-    @Test
-    public void parse_invalidZoomLink_failure() {
-        //String invalidLink = " " + PREFIX_NAME + "CS2103T" + " " + PREFIX_ZOOM_LINK + "1234";
-        //assertParseFailure(parser, invalidLink, ZoomLink.MESSAGE_CONSTRAINTS);
+        String invalidName = " " + PREFIX_NAME + "@@@@@@@ " + PREFIX_MODULAR_CREDITS + VALID_MC_4
+                + " " + PREFIX_TAG + VALID_TAG_LECTURE;
+        assertParseFailure(parser, invalidName, ModuleName.MESSAGE_CONSTRAINTS);
     }
 
     @Test
     public void parse_invalidMC_failure() {
-        //String invalidMC = " " + PREFIX_NAME + "CS2103T"
-        //        + " " + PREFIX_ZOOM_LINK + "www.example.com"
-        //        + " " + PREFIX_MODULAR_CREDITS + "3.9";
-        //assertParseFailure(parser, invalidMC, ModularCredits.MESSAGE_CONSTRAINTS);
+        String invalidMC = " " + PREFIX_NAME + VALID_MODULENAME_CS2103T + " " + PREFIX_MODULAR_CREDITS + "-2"
+                + " " + PREFIX_TAG + VALID_TAG_LECTURE;
+        assertParseFailure(parser, invalidMC, ModularCredits.MESSAGE_CONSTRAINTS);
     }
 
     @Test
     public void parse_invalidTags_failure() {
-        //String invalidTags = ""; // need to implement this.
-        //assertParseFailure(parser, invalidTags, Tag.MESSAGE_CONSTRAINTS);
+        String invalidTags = " " + PREFIX_NAME + VALID_MODULENAME_CS2103T + " " + PREFIX_MODULAR_CREDITS + "-2"
```

Comment:
Since we are testing cases with invalid tags, I think the modular credits argument should be valid?